packages/medusa/src/services/product-category.ts

```diff
@@ -0,0 +1,67 @@
+import { isDefined, MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+import { TransactionBaseService } from "../interfaces"
+import { ProductCategory } from "../models"
+import { ProductCategoryRepository } from "../repositories/product-category"
+import { FindConfig, Selector } from "../types/common"
+import { buildQuery } from "../utils"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  productCategoryRepository: typeof ProductCategoryRepository
+}
+
+/**
+ * Provides layer to manipulate product categories.
+ */
+class ProductCategoryService extends TransactionBaseService {
+  protected manager_: EntityManager
+  protected readonly productCategoryRepository_: typeof ProductCategoryRepository
+  protected transactionManager_: EntityManager | undefined
+
+  constructor({
+    manager,
+    productCategoryRepository,
+  }: InjectedDependencies) {
+    super(arguments[0])
```

Comment:
The linter should complain here as well 😄 
```suggestion
    // eslint-disable-next-line prefer-rest-params
    super(arguments[0])
```